apps/web/src/app/api/auth/google/start/route.ts

```diff
@@ -0,0 +1,39 @@
+import { googleProvider } from '$/services/auth'
+import { env } from '@latitude-data/env'
+import { generateCodeVerifier, generateState } from 'arctic'
+import { cookies } from 'next/headers'
+import { NextResponse } from 'next/server'
+
+export async function GET(): Promise<Response> {
+  const state = generateState()
+  const codeVerifier = generateCodeVerifier()
+  const url = googleProvider.createAuthorizationURL(state, codeVerifier, [
+    'openid',
+    'profile',
+    'email',
+  ])
+
+  const cookieStore = await cookies()
+
+  cookieStore.set('google_oauth_state', state, {
+    path: '/',
+    httpOnly: true,
+    secure: env.NODE_ENV === 'production',
+    maxAge: 60 * 10, // 10 minutes
+    sameSite: 'lax',
+  })
+  cookieStore.set('google_code_verifier', codeVerifier, {
+    path: '/',
+    httpOnly: true,
+    secure: env.NODE_ENV === 'production',
+    maxAge: 60 * 10, // 10 minutes
+    sameSite: 'lax',
+  })
```

Comment:
All this cookie management put in the same place we have the rest

Reply:
why? no i think thios is fine here it's just a temporary cookie for the google auth dance it does not make sense to abstract it to a service as it will never be reused